src/firebolt/async_db/cursor.py

```diff
@@ -267,6 +272,13 @@ def _reset(self) -> None:
         self._idx = 0
         self._row_sets = []
         self._next_set_idx = 0
+        self._query_id = ""
+
+    def _query_id_from_response_async(self, response: Response) -> str:
+        if response.headers.get("content-length", "") == "0":
+            raise InternalError("No response to asynchronous query.")
+        query_data = response.json(parse_float=str)
+        return query_data["query_id"]
```

Comment:
Validate that `query_id` is in `query_data`

Reply:
Done. I made it an `OperationalError` if it's missing.

Reply:
Also, a test for this error will also be added, https://github.com/firebolt-db/firebolt-python-sdk/pull/187#discussion_r930235178